handsontable/src/core.js

```diff
@@ -1395,6 +1399,34 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
     }
   }
 
+  /**
+   * Ensure that in case of the table being initialized with `display: none`, the table will get re-rendered after
+   * switching the `display` property to anything visible.
+   *
+   * @param {boolean|Array} isInit Indicator of whether the table is being initialized.
+   */
+  function correctFirstVisibleRender(isInit) {
```

Comment:
The core.js is big enough, so I'd be for moving that function to a separate file. By doing that at the same time, I'd propose making the function more generic. I'm thinking about a function that triggers a callback when the visibility of the observed element is changed. Something like:
```js
function observeVisibilityChangeOnce(elementToObserve, callback) {
  const observer = new IntersectionObserver((entries, observer) => {
    entries.forEach((entry) => {
      if (entry.isIntersecting && elementToObserve.offsetParent !== null) {
        callback();
        observer.unobserve(elementToObserve);
      }
    });
  }, {
    root: elementToObserve.ownerDocument.body
  });

  observer.observe(elementToObserve);
}
```
and then in the Core.js usage may look like this:
```js
if (!!firstRun && instance.rootElement.offsetParent === null) {
  observeVisibilityChangeOnce(instance.rootElement, () => instance.render());
}
```

Reply:
Good idea 👍